frontend/app/actions/ActionRefiners.scala

```diff
@@ -33,20 +42,73 @@ object Functions extends LazyLogging {
   def authenticated(onUnauthenticated: RequestHeader => Result = chooseRegister(_)): ActionBuilder[AuthRequest] =
     new AuthenticatedBuilder(AuthenticationService.authenticatedUserFor, onUnauthenticated)
 
-  def memberRefiner(onNonMember: RequestHeader => Result = notYetAMemberOn(_)) =
-    new ActionRefiner[AuthRequest, AnyMemberTierRequest] {
-      override def refine[A](request: AuthRequest[A]) = request.forMemberOpt {
-        _.map(member => MemberRequest(member, request)).toRight(onNonMember(request))
+  type SubRequestOrResult[A] = Future[Either[Result, SubReqWithSub[A]]]
+  type PaidSubRequestOrResult[A] = Future[Either[Result, SubscriptionRequest[A] with PaidSubscriber]]
```

Comment:
Wouldn't it be easier to work with the types:

```scala
type EitherResult[T] = Either[Result, T]
type FutureEitherResult[A] = EitherT[Future, Result, A]
```

so that `getSubRequest` takes a `Result` as argument and returns a `FutureEitherResult[A]`. That way we don't have to do this gymnastic of going from option to either, and of going out of the monad transformer prematurily